src/Illuminate/Mail/Mailer.php

```diff
@@ -553,41 +546,19 @@ protected function dispatchSentEvent($message, $data = [])
     {
         if ($this->events) {
             $this->events->dispatch(
-                new MessageSent($message->getSwiftMessage(), $data)
+                new MessageSent($message->getSymfonyMessage(), $data)
             );
         }
     }
 
     /**
-     * Force the transport to re-connect.
-     *
-     * This will prevent errors in daemon queue situations.
-     *
-     * @return void
-     */
-    protected function forceReconnection()
-    {
-        $this->getSwiftMailer()->getTransport()->stop();
-    }
```

Comment:
Starting a new discussion because the old one is a little bit hidden: https://github.com/laravel/framework/pull/38481#discussion_r696621855

@taylorotwell I would like to move the discussion regarding the forceReconnection to this thread.

Comment by @taylorotwell:
@driesvints @Jubeki can we revisit the force reconnection stuff? Was there a longer discussion of that somewhere? Do we have a good grasp over why we did that in the past and the ramifications of removing that behavior?

Reply:
Well there are basically three options:

1. Make the `stop` method available in Symfony (API Drivers don't really have a stop method though, they are only available for SMTP Drivers: https://github.com/symfony/symfony/blob/96c2e10395d3e699c8a3cbced69bf77a0e55a7fc/src/Symfony/Component/Mailer/Transport/Smtp/SmtpTransport.php#L260-L276)

2. Remove forceReconnection() from Mailer and add it to MailManager which will create a new Transport. This will lead to a forced Reconnection. I don't know if it is possible to access the MailManager from the Mailer.

3. Write all transports from scratch and add the stop method (I would like to avoid that if possible)

Reply:
forcedReconnection was introduced in Laravel 5.2 with the following PR: #13583

The Problem was that long working processes can throw SSL-Errors when trying to send. I didn't quite get the reason why though.

Reply:
I do see that the Symfony Mailer will internally "ping" the SMTP server if the last message was sent after a certain threshold time. It will then "stop" the SmtpTransport if it encounters an error during that ping.

We may just have to ship it as is and see what pops up in this regard. It looks like Symfony Mailer may have some handling in this regard that maybe SwiftMailer didn't have?

Reply:
Yeah I don't see any usage of the ping method in Swift_Mailer. The method exists and it seems to be called from the failover transport and the load balanced transport.

I think that pinging after some time has elapsed will solve the Problem. You can also set the threshold lower if necessary. Which is almost like a very frequent forcedReconnection.

Reply:
Yeah I believe this should be covered by Symfony Mailer now as @Jubeki explains albeit differently.